components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1931,7 +1935,10 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         await this.guardTeamOperation(teamId, "update");
         const team = await this.teamDB.findTeamById(teamId);
         try {
-            const customer = await this.stripeService.createCustomerForTeam(user, team!, setupIntentId);
+            let customer = await this.stripeService.findCustomerByTeamId(team!.id);
+            if (!customer) {
+                customer = await this.stripeService.createCustomerForTeam(user, team!, setupIntentId);
+            }
```

Comment:
It seems it is possible for 2 parallel requests to `subscribeTeamToStripe` to succeed and create 2 different subscriptions. Is this somehow transactional on the Stripe side?

Reply:
Thanks for identifying this pre-existing problem! I don't think it can realistically happen, but I'm happy to open a follow-up issue about this if it seems important. 💡

Reply:
The controller already handles this case - https://github.com/gitpod-io/gitpod/pull/10854/files#diff-1f1047abea4ed15decb31dfdf94d84646a601fbacd5462dcef20f7f8b4de33dbR54 it will ignore any with multiple subscriptions and logs an error.